app/core/Authentication/Authentication.ts

```diff
@@ -105,7 +112,7 @@ class AuthenticationService {
     await KeyringController.submitPassword(password);
     const encryptionLib = await AsyncStorage.getItem(ENCRYPTION_LIB);
     const existingUser = await AsyncStorage.getItem(EXISTING_USER);
-    if (encryptionLib !== ORIGINAL && existingUser) {
+    if (!encryptionLib && existingUser) {
```

Comment:
Hmm, shouldn't this still be `encryptionLib !== ORIGINAL` ?

```suggestion
    if (encryptionLib !== ORIGINAL && existingUser) {
```

We want to recreate the vault if the encryption library used was anything other than the one we expect, which is `ORIGINAL`.

Though I know we aren't aware of any cases where this is set to any other value, I didn't go digging in the git history to search for such examples, so leaving this as-is seemed safer